hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -683,6 +683,16 @@ private boolean checkIfJoinRequestFromAnExistingMember(JoinMessage joinMessage,
             return true;
         }
 
+        // TODO check if the TcpServerControl.process0(TcpServerConnection, Address, Collection<Address>, MemberHandshake)
+        // could be a better place for the following validation
```

Comment:
I agree it's best to catch this pathological condition immediately on member handshake.